src/main.cpp

```diff
@@ -18,48 +21,34 @@ static std::vector<std::vector<std::string>> levels;
 
 extern "C" {
 void sokoban_initialize() {
-    levels = {
-        {
-            // greg messing around
-            "   ####",
-            "####. ##",
-            "#.##$ .##",
-            "# ##    #",
-            "#    $  #",
-            "# #### ##",
-            "#@ $   #",
-            "#####  #",
-            "    ####",
-        },
-        {
-            // GRIGoRusha: Shito-Krito #86
-            "  #########",
-            "  #   #  @#",
-            " ##  .$.# #",
-            " #  ##  $ #",
-            "##  .$*#$##",
-            "#  #$#   # ",
-            "#  . .# ## ",
-            "####    #  ",
-            "   ######  ",
-        },
-        {
-            "######",
-            "#@ $.#",
-            "######",
-        },
-        {
-            "######",
-            "#@ $.#",
-            "######",
-        },
-        {
-            "######",
-            "#@ $.#",
-            "######",
-        },
+    std::filesystem::path getLevelsDir = std::filesystem::directory_entry("src/assets/levels");
+    std::vector<std::filesystem::path> get_map_path; // save path into vector element
+
+    for (const auto& entry : std::filesystem::directory_iterator(getLevelsDir)) {
+        get_map_path.push_back(entry.path());
+    }
+
+    std::cout << "LOOK HERE: " << get_map_path.size() << std::endl;
+
+    // opening map files stored in vector directory
+    for (unsigned int i = 0; i < get_map_path.size(); i++) {
+        std::ifstream openMapFile(get_map_path[i].c_str());
+        std::string store_line;
+        std::vector<std::string> storeMap;
+
+        if (!openMapFile) {
+            std::cerr << "Cannot open the File : " << get_map_path[i] << std::endl;
+        }
+
+        while (std::getline(openMapFile, store_line)) {
+            if (std::regex_match(store_line, std::regex("[@$*.+]+[#]{2,}"))) {
```

Comment:
This regex seems incorrect. Has it been tested? The regex here says "match one or more non-wall soko characters, then match two or more consecutive `'#'` characters". So this won't match valid soko lines like `"# @ #"`.

`[#]{2,}` can simplify to `#{2,}`, the brackets aren't useful when there's only one item between them.

I'd suggest

```
if (std::regex_match(line, "^[@ .$#+]+$")) {...}
```

This says "validate that the line contains nothing but Sokoban characters". I haven't tested it so make sure it works on our level files.